third_party/sol2/include/sol/sol.hpp

```diff
@@ -19416,7 +19416,13 @@ namespace sol { namespace function_detail {
 		}
 
 		template <bool is_yielding, bool no_trampoline>
-		static int call(lua_State* L) noexcept(std::is_nothrow_copy_assignable_v<T>) {
+		static int call(lua_State* L) 
+#if SOL_IS_ON(SOL_COMPILER_CLANG)
```

Comment:
Took this workaround from https://github.com/ThePhD/sol2/issues/1581#issuecomment-2103463524

I don't see another way to build with newer clang unfortunately. I'll try to make a PR to have this workaround in sol2 mainline...

Reply:
Consider adding a link as a comment right in the source. This way one can trace the origin of this bandaid, and eventually remove it in the future.